transport/src/main/java/io/netty/channel/oio/OioByteStreamChannel.java

```diff
@@ -126,25 +126,19 @@ protected void doWriteFileRegion(FileRegion region) throws Exception {
             outChannel = Channels.newChannel(os);
         }
 
-        long written = 0;
-        for (;;) {
-            long localWritten = region.transferTo(outChannel, written);
-            if (localWritten == -1) {
-                checkEOF(region);
+        long bytesToTransfer = obj.readableBytes();
+        long totalTransferred = 0;
+        while (obj.isReadable()) {
+            long transferred = obj.readTo(outChannel, obj.readableBytes());
+            if (transferred == -1) {
```

Comment:
For `FileChannel.transferTo`, return `0` does not mean `EOF` I think, it only means it does not transfer any bytes. I think we need to check our `position` and `FileChannel.size()` somewhere to expose `EOF`?